models/modules/resnet_architecture/mobile_resnet_generator.py

```diff
@@ -80,11 +80,21 @@ def __init__(self, input_nc, output_nc, ngf, norm_layer=nn.InstanceNorm2d,
                  wplus, init_type, init_gain, gpu_ids,
                  img_size, img_size_dec)
                 
-    def forward(self, input):
-        """Standard forward"""
-        output = self.encoder(input)
-        output = self.decoder(output)
-        return output
+    def forward(self, input, layers=[], encode_only=False):
+        if -1 in layers:
```

Comment:
Not sure I understand this...

Reply:
by putting '-1' into layers, the model will return features just after the last layer (it's the same as the encoder's output)

Reply:
So if layers is `[..., -1, ...]` you transform it to `[..., -1, ..., len(self.encoder)]` right? Is that the intended behavior? It looks strange to me

Reply:
yes because the forward will only check if the nth layer is in `layers`